EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -1380,6 +1383,11 @@ void V2::World::addUnions()
 			{
 				auto unionCores = culturalUnionMapper->getCoresForCulture(culture);
 				auto nationalCores = culturalNationalitiesMapper->getCoresForCulture(culture);
+
+				// Skip Emperor union if the Empire was decentralized
```

Comment:
if you reverse those 2 lines and preemptively clear emperor's union tag, you don't need these lines at all.